pkg/operator/controller/ingress/controller_test.go

```diff
@@ -1710,6 +1737,34 @@ func Test_computeUpdatedInfraFromService(t *testing.T) {
 			expectUpdated: true,
 			expectError:   false,
 		},
+		{
+			description:   "aws platform without DNSType set",
+			platform:      &awsPlatform,
+			ingresses:     []corev1.LoadBalancerIngress{},
+			expectUpdated: false,
+			expectError:   false,
+		},
+		{
+			description:   "aws platform with DNSType and no LB IP",
+			platform:      &awsPlatformWithDNSType,
+			ingresses:     []corev1.LoadBalancerIngress{},
+			expectUpdated: false,
+			expectError:   false,
+		},
+		{
+			description:   "aws platform with DNSType and no LB IP in infra config, service has 1 IP",
+			platform:      &awsPlatformWithDNSType,
+			ingresses:     awsIngresses,
+			expectUpdated: true,
+			expectError:   false,
+		},
+		{
+			description:   "aws platform with no change to LB IPs",
+			platform:      &awsPlatformWithLBIP,
+			ingresses:     awsIngresses,
+			expectUpdated: false,
+			expectError:   false,
+		},
 	}
 	for _, tc := range testCases {
 		t.Run(tc.description, func(t *testing.T) {
```

Comment:
Take a look at https://github.com/openshift/cluster-ingress-operator/pull/1167/commits/90693b70c8ddc4e2a8a815968d970d0817e33734#diff-bc23e1b7c8999c0b9674840022c285029528b1777b0fab5cc284974dfaa9cbebR1794, the unit test is still only checking GCP:
```
if !slices.Contains(infraConfig.Status.PlatformStatus.GCP.CloudLoadBalancerConfig.ClusterHosted.IngressLoadBalancerIPs, configv1.IP(ingress.IP)) {
```

Reply:
Also, we are still assuming IP and not hostname [here](https://github.com/openshift/cluster-ingress-operator/pull/1167/commits/90693b70c8ddc4e2a8a815968d970d0817e33734#diff-bc23e1b7c8999c0b9674840022c285029528b1777b0fab5cc284974dfaa9cbebR1793):
```
if len(ingress.IP) > 0 {
```